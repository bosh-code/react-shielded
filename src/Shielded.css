.button {
  appearance: none;
  background: none;
  display: inline-block;
  border: none;
  margin: 0;
  padding: 0;
  cursor: pointer;
}
