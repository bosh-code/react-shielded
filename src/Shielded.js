import React, { useRef, useState } from 'react';
import Modal from './Modal';
import styles from './Shielded.css';
import Button from './button.svg';

const Shielded = () => {
  const buttonRef = useRef(null);

  const [isOpen, setOpen] = useState(false);

  return (
    <>
      <button
        className={styles.button}
        onClick={() => setOpen(true)}
        ref={buttonRef}
        style={{ opacity: isOpen ? 0 : 1 }}
        type="button"
      >
        <Button />
      </button>
      {isOpen && (
        <Modal
          onRequestClose={() => {
            setOpen(false);
            buttonRef.current.focus();
          }}
        />
      )}
    </>
  );
};

Shielded.propTypes = {};

export default Shielded;
