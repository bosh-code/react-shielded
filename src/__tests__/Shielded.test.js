import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import Shielded from '../Shielded';

test('renders and matches snapshot', () => {
  const { container } = render(<Shielded />);
  expect(container.firstChild).toMatchSnapshot();
});

test('clicking button opens modal', () => {
  const { getByLabelText, getByRole } = render(<Shielded />);
  userEvent.click(getByLabelText('Women’s Refuge'));
  expect(getByRole('dialog')).toBeTruthy();
});

test('closing modal focuses button', () => {
  const { getByLabelText } = render(<Shielded />);
  const button = getByLabelText('Women’s Refuge');
  userEvent.click(button);
  userEvent.click(getByLabelText('Close'));
  expect(button).toHaveFocus();
});
