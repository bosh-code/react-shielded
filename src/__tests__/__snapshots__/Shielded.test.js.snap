// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches snapshot 1`] = `
<button
  aria-label="Womenâ€™s Refuge"
  style="display: inline-block; margin: 0px; padding: 0px; appearance: none; background: none; cursor: pointer; opacity: 1;"
  type="button"
>
  <svg />
</button>
`;
